pkg/util/log: add metrics to otlp sink

TheComputerM opened the pull request:
adds metrics to monitor internal statistics of otlp sinks, similar to
the metrics exposed for fluent sinks.

`log.otlp.sink.write.attempts`: Number of write attempts experienced by
otlp-server logging sinks

`log.otlp.sink.write.errors`: Number of write errors experienced by
otlp-server logging sinks

`log.otlp.sink.grpc.transparent_retries`: Number of transparent retries
done by otlp-server logging sinks when using GRPC

Epic: none
Release note: none

---

**Happy case**

<img width="1133" height="238" alt="image" src="https://github.com/user-attachments/assets/1c03c159-e88e-42e1-9e09-e4a66a5f2fe1" />

**When OTLP receiver is down**

<img width="1133" height="238" alt="image" src="https://github.com/user-attachments/assets/3cc59391-f6be-4452-b4df-aff2606b34c8" />

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/150622)
<!-- Reviewable:end -->

TheComputerM commented:
It isn't possible for some fluent metrics (like `FluentSinkConnectionAttempt` and `FluentSinkConnectionError`) to have a direct counterpart because gRPC hides the underlying connection details and only exposes RPC metrics: https://grpc.io/docs/guides/opentelemetry-metrics/#per-call-metrics

TheComputerM committed:
pkg/util/log: add metrics to otlp sink

adds metrics to monitor internal statistics of otlp sinks, similar to
the metrics exposed for fluent sinks.

log.otlp.sink.write.attempts: Number of write attempts experienced by
otlp-server logging sinks

log.otlp.sink.write.errors: Number of write errors experienced by
otlp-server logging sinks

log.otlp.sink.grpc.transparent_retries: Number of transparent retries
done by otlp-server logging sinks when using GRPC

Epic: none

Release note: none

arjunmahishi reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/150622#-:-OVpXKAnCCCd-m_xLnMD:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @aa-joshi, @Abhinav1299, and @alyshanjahani-crl)



<!-- Sent from Reviewable.io -->

TheComputerM commented:
bors r+
